[BUG] Building CLion project with MVSC on Windows 10 fails since sep 5, 2020

## Issue description

Building a CLion project with pybind11 on Windows 10 with the MVSC toolchain fails since sep 5.
Instead of succesfullly building a debug and release Python module, both build types now fail.
The release build fails with:
```
[100%] Linking CXX shared module spam
LINK: command "C:\PROGRA~2\MICROS~1\2019\BUILDT~1\VC\Tools\MSVC\1423~1.281\bin\Hostx64\x64\link.exe /nologo @CMakeFiles\spam.dir\objects1.rsp /out:spam /implib:spam.lib /pdb:D:\GitHub\pybind11-issue\cmake-build-release\spam.pdb /dll /version:0.0 -LTCG /machine:x64 /INCREMENTAL:NO spamlib.lib C:\Users\Ruud\AppData\Local\Programs\Python\Python38\libs\python38.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:spam.manifest" failed (exit code 1120) with the following output:
   Creating library spam.lib and object spam.exp
spam.cpp.obj : error LNK2001: unresolved external symbol __imp__Py_NegativeRefcount
spam.cpp.obj : error LNK2001: unresolved external symbol __imp__Py_RefTotal
spam : fatal error LNK1120: 2 unresolved externals
NMAKE : fatal error U1077: 'C:\Users\Ruud\AppData\Local\JetBrains\Toolbox\apps\CLion\ch-0\202.7319.72\bin\cmake\win\bin\cmake.exe' : return code '0xffffffff'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.23.28105\bin\HostX64\x64\nmake.exe"' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.23.28105\bin\HostX64\x64\nmake.exe"' : return code '0x2'
Stop.
```

The debug build fails with:
```
[ 75%] Linking CXX shared module spam
LINK Pass 1: command "C:\PROGRA~2\MICROS~1\2019\BUILDT~1\VC\Tools\MSVC\1423~1.281\bin\Hostx64\x64\link.exe /nologo @CMakeFiles\spam.dir\objects1.rsp /out:spam /implib:spam.lib /pdb:D:\GitHub\pybind11-issue\cmake-build-debug\spam.pdb /dll /version:0.0 /machine:x64 /debug /INCREMENTAL spamlib.lib C:\Users\Ruud\AppData\Local\Programs\Python\Python38\libs\python38.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:CMakeFiles\spam.dir/intermediate.manifest CMakeFiles\spam.dir/manifest.res" failed (exit code 1104) with the following output:
LINK : fatal error LNK1104: cannot open file 'python38_d.lib'
NMAKE : fatal error U1077: 'C:\Users\Ruud\AppData\Local\JetBrains\Toolbox\apps\CLion\ch-0\202.7319.72\bin\cmake\win\bin\cmake.exe' : return code '0xffffffff'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.23.28105\bin\HostX64\x64\nmake.exe"' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.23.28105\bin\HostX64\x64\nmake.exe"' : return code '0x2'
Stop.

```

## Reproducible example code
A minimal example of the project is provided in https://github.com/rhjdjong/pybind11-issue.
The environment used for this project is CLion 2020.2 with the following toolchain:
* MVSC 2019 BuildTools Version 16.0, Architecture amd64
* CMake (bundled with CLion) version 3.17.3
* Python 3.8.0 64-bit standard window install
* Pybind11 cloned into a neighbouring directory.

By resetting  the head of my local installed pybind11 repo, I discovered that this problem started with revision ce1a07ef45d204e6e72d7039ca6764cac90b026e. By looking at the changes and some experimenting, I found that lines 74-76 in tools/pybind11NewTools.cmake triggered these errors. This revision changed that line from
```
execute_process(COMMAND ${_Python}::Python -c "import sys; print(hasattr(sys, 'gettotalrefcount'))"
                OUTPUT_VARIABLE PYTHON_IS_DEBUG)
```
into
```
execute_process(
  COMMAND "${${_Python}_EXECUTABLE}" "-c" "import sys; sys.exit(hasattr(sys, 'gettotalrefcount'))"
  RESULT_VARIABLE PYTHON_IS_DEBUG)
```
I verified that changing this back to the original line makes the builds succeed again, also with the current pybind11 master HEAD (revision 5e6ec496522b313e34af3de91f6c0565f68e3552).

--- a/tools/pybind11NewTools.cmake
+++ b/tools/pybind11NewTools.cmake
@@ -70,6 +70,16 @@ if(PYBIND11_MASTER_PROJECT)
   endif()
 endif()
 
+# If a user finds Python, they may forget to include the Interpreter component
+# and the following two steps require it. It is highly recommended by CMake
+# when finding development libraries anyway, so we will require it.
+if(NOT DEFINED ${_Python}_EXECUTABLE)
+  message(
+    FATAL_ERROR
+      "${_Python} was found without the Interpreter component. Pybind11 requires this component.")
+
+endif()
+
 # Debug check - see https://stackoverflow.com/questions/646518/python-how-to-detect-debug-Interpreter
 execute_process(
   COMMAND "${${_Python}_EXECUTABLE}" "-c" "import sys; sys.exit(hasattr(sys, 'gettotalrefcount'))"

